### PR TITLE
py-multiqc: add v1.23

### PR DESCRIPTION
**Pull request by vladsavelyev**
Updated `py-multiqc` to 1.23.

Adding myself and @ewels as maintainers.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-multiqc/package.py [hunk 1]
```diff
@@ -34,7 +36,6 @@ class PyMultiqc(PythonPackage):
     depends_on("py-jinja2@2.9:", type=("build", "run"), when="@:1.13")
     depends_on("py-kaleido", type=("build", "run"), when="@1.20:")
     depends_on("py-markdown", type=("build", "run"), when="@1.3:")
-    depends_on("py-matplotlib@2.1.1:", type=("build", "run"), when="@1.13:")
```
Comment (ewels) [inline var/spack/repos/builtin/packages/py-multiqc/package.py:37]: Do we need to make it `when=":@1.21"` rather than flat out deleting this line?

**Review by ewels — APPROVED** [review]
Thanks!

File: var/spack/repos/builtin/packages/py-multiqc/package.py (hunk 1 above)
Comment (vladsavelyev) [reply var/spack/repos/builtin/packages/py-multiqc/package.py:37]: We can delete old versions, of course

**Comment by vladsavelyev** [comment]
@ewels I'm not allowed to merge. Can you please?

**Review by ewels — APPROVED** [review]

**Comment by ewels** [comment]
Looks like I don't have permission yet either. I guess that might kick in once this is merged?

**Review by bernhardkaindl — APPROVED** [review]
@vladsavelyev @ewels Approving as you are the upstream authors/maintainers.

Normally, `spack` tries to normally not remove old versions if they are used by other recipes.

But in this case, no other upstream `spackage` appears to use it.

To get merge-to-master permissions for `spack`, you have to become a `spack` maintainer. Spack needs maintainers, but you'd have like to contribute to `spack` by reviewing other PRs besides your package as well.

**Pull request merged**